scripts/globals/abilities/shadowbind.lua

```diff
@@ -14,15 +14,33 @@ require("scripts/globals/status");
 -----------------------------------
 
 function onAbilityCheck(player,target,ability)
-    return 0,0;
+    if ((player:getWeaponSkillType(SLOT_RANGED) == SKILL_MRK and player:getWeaponSkillType(SLOT_AMMO) == SKILL_MRK) or
+    (player:getWeaponSkillType(SLOT_RANGED) == SKILL_ARC and player:getWeaponSkillType(SLOT_AMMO) == SKILL_ARC)) then
+        return 0,0;
+    end
+    return 216,0; -- You do not have an appropriate ranged weapon equipped.
 end;
 
 -----------------------------------
 -- onUseAbility
 -----------------------------------
 
-function onUseAbility(player,target,ability)
+function onUseAbility(player,target,ability,action)
+
+    local duration = 30 + player:getMod(MOD_SHADOW_BIND_EXT);
+
+     -- TODO: Acc penalty for /RNG, acc vs. mob level?
+    if (math.random(0, 100) >= target:getMod(MOD_BINDRES) and target:hasStatusEffect(EFFECT_BIND) == false) then
+        target:addStatusEffect(EFFECT_BIND, 0, 0, duration);
+        ability:setMsg(277); -- Target is bound.
+	else
+        ability:setMsg(158); -- Player uses Shadowbind, but misses.
+    end
+
+    local del = player:delItem(player:getEquipID(SLOT_AMMO), 1); -- deplete one round of ammo
```

Comment:
https://github.com/DarkstarProject/darkstar/commit/94db83a313f68cfe6cd7545e2f75b09f105af9c4 yep that's the one.